itests/org.openhab.binding.mqtt.homie.tests/src/main/java/org/openhab/binding/mqtt/HomieImplementationTest.java

```diff
@@ -164,16 +181,18 @@ public void tearDown() throws InterruptedException, ExecutionException, TimeoutE
 
     @Test
     public void retrieveAllTopics() throws InterruptedException, ExecutionException, TimeoutException {
-        CountDownLatch c = new CountDownLatch(registeredTopics);
-        connection.subscribe(DEVICE_TOPIC + "/#", (topic, payload) -> c.countDown()).get(200, TimeUnit.MILLISECONDS);
-        assertTrue("Connection " + connection.getClientId() + " not retrieving all topics",
-                c.await(1000, TimeUnit.MILLISECONDS));
+        // four topics are not under /testnode !
+        CountDownLatch c = new CountDownLatch(registeredTopics - 4);
```

Comment:
Hm, with the constant "4" this test class is quite fragile towards extendability. Maybe repurpose the variable only count "/testnode" topics?

Reply:
I would like to split off the moquette-thing anyway, it's  not directly related. I merely pushed it to show that it's working. I'll check how this can be changed.

Reply:
Looks like the constant is still there. Is it easy to fix or should we merge it as-is?

Reply:
It's not so easy to fix. I think that the comment will help if someone changes the test and it fails again.

Reply:
OK let's then merge it! The comment it certainly helpful.